src/distilabel/steps/task/self_instruct.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from typing import TYPE_CHECKING, Any, Dict, List, Union
+
+from jinja2 import Template
+from pydantic import PrivateAttr
+
+from distilabel.steps.task.base import Task
+
+if TYPE_CHECKING:
+    from distilabel.steps.task.typing import ChatType
+
+_SELF_INSTRUCT_TEMPLATE = """
+# Task Description
+Develop {{ num_instructions }} user queries that can be received by the given AI application and applicable to the provided context. Emphasize diversity in verbs and linguistic structures within the model's textual capabilities.
+
+# Criteria for Queries
+{{ criteria_for_query_generation }}
+Write each query on a separate line and avoid using numbered lists or bullet points.
+
+# AI Application
+{{ application_description }}
+
+# Context
+{{ input }}
+
+# Output
+"""
+
+_PARSE_OUTPUT_REGEX = re.compile(r"(?<=# Output\n)(.*)", re.IGNORECASE)
```

Comment:
Feel free to keep the regex defined within the `SelfInstruct` class, as this is not imported/used anywhere else

Reply:
Indeed AFAIK this is not even used within this file, right?